src/Pivot.jl

```diff
@@ -0,0 +1,87 @@
+│ 6   │ 3     │ 6     │ 9     │ cf_val1_3 │ cf_val2_3 │ c      │ f      │
+│ 7   │ 1     │ 4     │ 7     │ de_val1_1 │ de_val2_1 │ d      │ e      │
+│ 8   │ 1     │ 4     │ 7     │ df_val1_1 │ df_val2_1 │ d      │ f      │
+│ 9   │ 2     │ 5     │ 8     │ de_val1_2 │ de_val2_2 │ d      │ e      │
+│ 10  │ 2     │ 5     │ 8     │ df_val1_2 │ df_val2_2 │ d      │ f      │
+│ 11  │ 3     │ 6     │ 9     │ de_val1_3 │ de_val2_3 │ d      │ e      │
+│ 12  │ 3     │ 6     │ 9     │ df_val1_3 │ df_val2_3 │ d      │ f      │
+
+```
+pivot_wider(df, names_from = [:cname1,:cname2], values_from = [:val1,:val2])
+```
+3×11 DataFrame
+│ Row │ x     │ a     │ b     │ val1_c_e  │ val1_c_f  │ val1_d_e  │ val1_d_f  │ val2_c_e  │ val2_c_f  │ val2_d_e  │ val2_d_f  │
+│     │ Int64 │ Int64 │ Int64 │ String?   │ String?   │ String?   │ String?   │ String?   │ String?   │ String?   │ String?   │
+├─────┼───────┼───────┼───────┼───────────┼───────────┼───────────┼───────────┼───────────┼───────────┼───────────┼───────────┤
+│ 1   │ 1     │ 4     │ 7     │ ce_val1_1 │ cf_val1_1 │ de_val1_1 │ df_val1_1 │ ce_val2_1 │ cf_val2_1 │ de_val2_1 │ df_val2_1 │
+│ 2   │ 2     │ 5     │ 8     │ ce_val1_2 │ cf_val1_2 │ de_val1_2 │ df_val1_2 │ ce_val2_2 │ cf_val2_2 │ de_val2_2 │ df_val2_2 │
+│ 3   │ 3     │ 6     │ 9     │ ce_val1_3 │ cf_val1_3 │ de_val1_3 │ df_val1_3 │ ce_val2_3 │ cf_val2_3 │ de_val2_3 │ df_val2_3 │
+"""
+function pivot_wider(df::AbstractDataFrame; names_from = nothing, values_from = nothing)
+   # A lot of branching in this code. I do not know if multiple dispatch can be done here efficiently. Not enough knowledge with multiple dispatch.
+
+   # Checking if a single column is provided or an array of columns either symbols or strings
+   if (names_from isa Symbol) || (names_from isa String)
+    names_from = [names_from]
+   end
+   if (values_from isa Symbol) || (values_from isa String)
+    values_from = [values_from]
+   end
+
+   ## Defining id_cols as all columns other than names and values columns
+   id_cols = setdiff(Symbol.(names(df)), names_from, values_from)
+   if names_from isa Array{String}
+      id_cols = setdiff(names(df), names_from, values_from)
+   end
+   
+   # Concatenating multiple name columns with "_" and creating a single name column cname_temp
+   # This may cause a bug if someone names their column cname_temp (Dont know how to work around this issue: creating a temporary column using it and then deleting it without conflicts)
+   df[!,:cname_temp] = [join(Array(r), "_") for r in eachrow(select(df, names_from))]
+   
+   # function to  unstack with column names added
+   unstack_name(df::AbstractDataFrame, rowkeys, colkey, value) = unstack(df::AbstractDataFrame, rowkeys, colkey, value, renamecols = x -> Symbol(string(value), "_", x))
+
+   # Broadcast unstacking on all value columns 
+   wide_dfs = unstack_name.(Ref(df), Ref(id_cols), Ref(:cname_temp), values_from)
```

Comment:
In that case, maybe just list comprehension is sufficient

```suggestion
   wide_dfs = [unstack(df, id_cols, :cname_temp, value, renamecols = x -> string(value)*"_"*x)
            for value in values_from]
```